### PR TITLE
Readme clarifications

### PR DESCRIPTION
**Pull request by milesfrain**
I believe the "Go to definition" issue is reported in #192

### CODE REVIEW

**Comment by milesfrain** [comment]
I'm also having trouble with the "Renaming via `HaRe`" feature.

I'm following the [HaRe installation instructions](https://github.com/alanz/HaRe#getting-started), but encountering an error:
```
$ stack install HaRe --resolver=lts-4.0
...
--  While building package happy-1.19.5 using:
      /tmp/stack-56b88fddfba6051c/happy-1.19.5/.stack-work/dist/x86_64-linux-tinfo6/Cabal-1.22.5.0/setup/setup --builddir=.stack-work/dist/x86_64-linux-tinfo6/Cabal-1.22.5.0 build --ghc-options ""
    Process exited with code: ExitFailure 1
Progress 54/65
```

I'd report this issue in the HaRe repo if issue reporting was available. It also seems like this statement doesn't align with the feature listed in this repo.
> Currently only emacs integration is offered.

**Comment by jneira** [comment]
Thanks a lot, they are all sensible updates. However i think "Go to definition" has different behaviours between haskell lang servers variants and all of them supports go to defs between files of the same component (and even between components, although not in a reliable way)

**Comment by milesfrain** [comment]
Is this expected to work with HIE backend?
f12

**Comment by parkerault** [comment]
> 
> 
> Is this expected to work with HIE backend?
> f12

I'm having the same issue. I see this hasn't been merged yet, so I was expecting to be able to jump to definition within the same project at least, since this is how the feature typically works with other language extensions.

**Comment by ndmitchell** [comment]
> Is this expected to work with HIE backend?

Nothing is expected to work with the HIE backend anymore. We strongly recommend moving to Haskell Language Server. There is no one left supporting HIE and all the features are being absorbed into HLS.

**Comment by milesfrain** [comment]
> Nothing is expected to work with the HIE backend anymore

So perhaps [this section](https://github.com/haskell/vscode-haskell/blob/master/README.md#language-servers) needs to be clarified in the readme. Maybe remove HIE as an option.
> Haskell IDE Engine: A **stable and mature** language server, but note that development has moved from this to the Haskell Language Server.

**Comment by ndmitchell** [comment]
I note that HIE was recently archived as a repo, so yes, I think it should be something along the lines of unsupported and legacy.

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Language server client for haskell using the [HIE](https://github.com/haskell/ha
 * Diagnostics via HLint and GHC warnings/errors
 * Code actions and quick-fixes via [`apply-refact`](https://github.com/mpickering/apply-refact) (click the lightbulb)
 * Type information and documentation (via hoogle) on hover
-* Jump to definition (`F12` or `Go to Definition` in command palette)
+* Jump to definition (`F12` or `Go to Definition` in command palette). Note that this only works for definitions within the current file.
```
Comment (bubba) [inline README.md:23]: I think this limitation is gone now in newer versions of ghcide (and hls)

Comment (jneira) [reply README.md:23]: But it still does not work for non local package definitions (i.e. libraries)

**Comment by bubba** [comment]
I've updated the readme to reflect that HIE is legacy

**Review by jneira — CHANGES_REQUESTED** [review]
Some minor changes to reflect actual state of things

**Comment by jneira** [comment]
@milesfrain sorry for the delay in the final review. As the first block changed to remove references to hie i had to choose the actual master version in the rebase. I've kept the clarifications in the debugging part.

Thanks!

**Review by jneira — APPROVED** [review]

**Pull request merged**